include/nlohmann/detail/iterators/iter_impl.hpp

```diff
@@ -118,6 +118,13 @@ class iter_impl
           to iterator is not defined.
     */
 
+    /* We had to explicitly define the copy constructor for the const
+       case to circumvent a bug on msvc 2015 debug build.
+       More info: https://github.com/nlohmann/json/issues/1608
+     */
+    iter_impl(const iter_impl<const BasicJsonType>& other) noexcept
```

Comment:
Could you please add some documentation like from the constructor below and add your note there as `@note` annotation?

Reply:
Sure! Done.

Let me know if you had any other suggestions on the wording, etc.